framework-tools-replay/src/main/java/uk/gov/justice/framework/tools/replay/database/LiquibaseFactory.java

```diff
@@ -1,24 +1,83 @@
 package uk.gov.justice.framework.tools.replay.database;
 
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLClassLoader;
+import java.nio.file.Path;
 import java.sql.SQLException;
 
+import javax.sql.DataSource;
+
 import liquibase.Liquibase;
 import liquibase.database.jvm.JdbcConnection;
 import liquibase.exception.LiquibaseException;
 import liquibase.resource.ClassLoaderResourceAccessor;
-import org.apache.commons.dbcp2.BasicDataSource;
 
+/**
+ * Factory for creating LiquibaseProxy
+ */
 public class LiquibaseFactory {
 
+    /**
+     * Create LiquibaseProxy with a liquibase script xml resource path and a given datasource.
+     *
+     * @param liquibaseChangeLog resource path of the liquibase change log to use
+     * @param dataSource         datasource to use for processing
+     * @return LiquibaseProxy that wraps a {@link Liquibase}
+     * @throws LiquibaseException thrown if Liquibase fails on creation
+     * @throws SQLException       thrown if Datasource fails to get a connection
+     */
     public LiquibaseProxy create(
-            final String liquibaseChangeLogXml,
-            final BasicDataSource dataSource) throws LiquibaseException, SQLException {
+            final String liquibaseChangeLog,
+            final DataSource dataSource) throws LiquibaseException, SQLException {
+
+        return liquibaseProxyWith(
+                liquibaseChangeLog,
+                new ClassLoaderResourceAccessor(parentClassloader()),
+                new JdbcConnection(dataSource.getConnection()));
```

Comment:
I had issues getting this to work previously, but have managed to get it to work now